src/entities/user/user.controller.ts

```diff
@@ -25,13 +28,53 @@ class UserController implements IController {
     }
 
     initializeRoutes(): void {
-        this.router.get(this.path, this.getSuggestions);
+        this.router.get(`${this.path}/validate_token`, getJWT, this.validateToken);
+        this.router.post(`${this.path}/login`, this.validator.body(valid.UserLoginSchema), this.login);
+        this.router.get(this.path, getJWT, this.getSuggestions);
         this.router.post(this.path, this.validator.body(valid.UserBodySchema), this.createUser);
-        this.router.get(`${this.path}/:id`, this.getUserById);
-        this.router.patch(`${this.path}/:id`, this.validator.body(valid.UserUpdateBodySchema), this.updateUser);
-        this.router.delete(`${this.path}/:id`, this.deleteUser);
+        this.router.get(`${this.path}/:id`, getJWT, this.getUserById);
+        this.router.patch(`${this.path}/:id`, getJWT, this.validator.body(valid.UserUpdateBodySchema), this.updateUser);
+        this.router.delete(`${this.path}/:id`, getJWT, this.deleteUser);
     }
 
+    validateToken = async (
+        _request: Request,
+        response: Response,
+        next: NextFunction,
+    ): Promise<void | Response<unknown, Record<string, unknown>>> => {
+        try {
+            Logger.info('token validated, user is Authorized.');
+
+            return response.status(200).json({
+                message: 'Authorized',
+            });
+        } catch (error) {
+            return next(error);
+        }
+    };
+
+    login = async (request: Request, response: Response, next: NextFunction): Promise<void> => {
+        try {
+            const { login, password } = request.body;
+            Logger.info(`${login} is attempting to login.`);
+            const user = await this.userService.loginUser(login, password);
+            if (user instanceof HttpException) throw user;
+            signJWT(user, (err, token) => {
```

Comment:
By rewriting signJWT utility without callbacks you'll make this method more readable and in-line with the rest of the codebase.

Reply:
done